src/rgw/rgw_common.cc

```diff
@@ -97,11 +97,48 @@ is_err() const
   return !(http_ret >= 200 && http_ret <= 399);
 }
 
+static bool starts_with(const string& s, const string& prefix) {
+  if (s.size() < prefix.size()) {
+    return false;
+  }
+  for (unsigned int i = 0; i < prefix.size(); ++i) {
+    if (prefix[i] != s[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// The requestURI transferred from the frontend can be abs_path or absoluteURI
+// If it is absoluteURI, we should adjust it to abs_path for the following 
+// S3 authorization and some other processes depending on the requestURI
+// The absoluteURI can start with "http://", "https://", "ws://" or "wss://"
+static string get_abs_path(const string& request_uri) {
+  const string ABS_PREFIXS[] = {"http://", "https://", "ws://", "wss://"};
```

Comment:
can we make this array static so we don't construct the std::strings on every call?